attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -3,17 +3,23 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use super::tsm_report::{provider_is, TsmReportPath, TsmReportProvider, TSM_REPORT_PROVIDER_TDX};
 use super::Attester;
 use anyhow::*;
 use base64::Engine;
 use serde::{Deserialize, Serialize};
 use std::path::Path;
+use std::result::Result::Ok;
+
+#[cfg(feature = "tdx-deprecated-getquote-ioctl")]
 use tdx_attest_rs;
 
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
 pub fn detect_platform() -> bool {
-    Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
+    provider_is(TSM_REPORT_PROVIDER_TDX)
+        || (cfg!(feature = "tdx-legacy-getquote-ioctl")
+            && (Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()))
```

Comment:
could you use `cfg-if!` to make the conditions more clearer?

Reply:
is that a good with here as there's no if-else needed?

Reply:
I'll rework on it a bit

Reply:
updated